sidecar/config/config.go

```diff
@@ -119,9 +125,10 @@ type HealthCheck struct {
 
 // Config stores the various configuration options for the sidecar
 type Config struct {
-	Register bool `yaml:"register"`
-	Proxy    bool `yaml:"proxy"`
-	DNS      bool `yaml:"dns"`
+	Register     bool   `yaml:"register"`
+	Proxy        bool   `yaml:"proxy"`
+	ProxyAdapter string `yaml:"proxy_adapter"`
```

Comment:
Can we move the proxyadapter under Proxy? as say, a "type" field ? This would allow us to add additional fields such as the port on which the proxy should be configured to listen on (something that we do not auto-configure now), and some other common proxy related settings.